lapce-data/src/document.rs

```diff
@@ -2217,6 +2236,40 @@ impl Document {
                 }
             }
         }
+        if let Some(bracket_styles) = self.parser.bracket_pos.get(&line) {
+            for bracket_style in bracket_styles.iter() {
+                if let Some(fg_color) = bracket_style.style.fg_color.as_ref() {
+                    if let Some(fg_color) = config.get_style_color(fg_color) {
+                        let start = phantom_text.col_at(bracket_style.start);
+                        let end = phantom_text.col_at(bracket_style.end);
+                        layout_builder = layout_builder.range_attribute(
+                            start..end,
+                            TextAttribute::TextColor(fg_color.clone()),
+                        );
+                    } else {
+                        let fg_color = match fg_color.as_str() {
+                            "bracket.color1" => {
+                                config.get_color_unchecked(LapceTheme::TERMINAL_BLUE)
+                            }
+                            "bracket.color2" => config
+                                .get_color_unchecked(LapceTheme::TERMINAL_YELLOW),
+                            "bracket.color3" => config
+                                .get_color_unchecked(LapceTheme::TERMINAL_MAGENTA),
+                            "bracket.unpaired" => {
+                                config.get_color_unchecked(LapceTheme::TERMINAL_RED)
+                            }
+                            _ => unreachable!(),
```

Comment:
this doesn't seem right, `TERMINAL` shouldn't be used in document

Reply:
I know it looks weird, but I can't find a good way to fallback to theme's basic color if the theme doesn't set bracket colorization colors. The basic color seems not loaded to the config.

Reply:
Fixed

Reply:
What I meant is that this shouldn't be matched by string, it should it's own theme key (since you added them to default configs anyway)

Reply:
Well the problem is if I download a theme without these keys, then there will be no color for brackets since `get_style_color` will not fallback to default theme.

Reply:
That is fine

Reply:
Fine, I removed it.